src/main/java/pl/edu/mimuw/logic/Not.java

```diff
@@ -0,0 +1,28 @@
+package pl.edu.mimuw.logic;
+
+public class Not extends Operator1Arg {
+
+  public Not(String name, Formula child) {
+    super(name, child);
+  }
+
+  @Override
+  public boolean eval(Valuation valuation) {
+    return !child.eval(valuation);
+  }
+
+  @Override
+  public Formula simplify() {
+    if (simpleChild.isConstant()) {
+      if (simpleChild.eval(null)) {
```

Comment:
lepiej byłoby to robić na pustym `Valuation` a nie na `null`, bo czytelniejsze, o co Ci chodzilo